### PR TITLE
Add StringIntentHandler

### PR DESCRIPTION
**Pull request by dreimer1986**
This one is made to pass over a complete string of the reply the user gives to the Echo device in question. I for example use this for chatting together with a Telegram bot. It ... works as long as the one on the other side talks in a way Alexa understands.

Sorry for the commit mess. I have NO clue how to squash this. GIT is a mess.

### CODE REVIEW

**Comment by DEADSEC-SECURITY** [comment]
Ill review this later this day and squash the PR

**Comment by DEADSEC-SECURITY** [comment]
Btw did you test this in your own alexa.

File: lambda/language_strings.json [hunk 1]
```diff
@@ -22,6 +23,7 @@
 		"ERROR_SPECIFIC_DATE": "Ich kann leider noch kein spezifisches Datum verarbeiten. Bitte nutze stattedessen Zeiträume wie beispielsweise... in 5 Stunden.",
 		"HELP_MESSAGE": "Dieser Skill sollte nur reaktiv genutzt werden, wenn er via home assistant angestossen wird",
 		"OKAY": "Okay",
+		"STRING" : "Du hast {} gewählt",
```
Comment (DEADSEC-SECURITY) [inline lambda/language_strings.json:26]: Please add this to all the languages.

**Review by DEADSEC-SECURITY — COMMENTED** [review]
Please add for the other languages too.

**Comment by dreimer1986** [comment]
> Btw did you test this in your own alexa.

Yes, I use it on my Show normally. That's where I found out that the Show behaves differently compared to the Dot I own now too.

File: lambda/language_strings.json (hunk 1 above)
Comment (dreimer1986) [reply lambda/language_strings.json:26]: Done

**Review by DEADSEC-SECURITY — CHANGES_REQUESTED** [review]
Please add to the skill-manifests different languages too

**Comment by dreimer1986** [comment]
I use it in two automations and the work fine with the recent version here:

```
- id: telegram-bot-to-alexa
  alias: Telegram Bot Textausgabe auf Alexa
  trigger:
  - platform: event
    event_type: telegram_text
  action:
  - parallel:
    - service: script.activate_alexa_actionable_notification
      data:
        text: '{{ trigger.event.data.text }}'
        event_id: alexa_notification_telegram-bot
        alexa_device: media_player.daniels_echo_show
    - service: notify.telegrambot
      data:
        message: '{{ trigger.event.data.text }}'
  mode: single
- id: alexa-to-telegram-bot
  alias: Alexa auf Telegram Bot Textausgabe
  trigger:
  - platform: event
    event_type: alexa_actionable_notification
    event_data:
      event_id: alexa_notification_telegram-bot
  action:
  - service: notify.telegrambot
    data:
      message: '{{ trigger.event.data.event_response }}'
  mode: single
```

Not much differs to the recent version here.

**Comment by rwunsch** [comment]
Thanks for the repy @dreimer1986 !

how do you trigger Alexa to use the "StringIntent" when responding to Alexa?
Maybe  I have a conceptual problem in understanding - but I thought the lambda-function/Alexa-service decides itself on the "intent to use".
So a human response with "yes" or "yeah, I would like that ..." would result in a YesIntent.

What would trigger this "custom StringIntent"? 
I was trying to respond with my hair-color, questions, short sentences and all kinds of other things (in german, but also tried in english, despite using the German interaction model).

I can trigger the "DataTimeIntent" why saying eg. "tomrrow at one pm". 
I have problems triggering the "Number"-Intent  as well (attempting to trigger the NumericIntentHandler) - it does not work with only responding with a number, eg. "42" (which worked in 0.8.2 for some reason - but I do not see why there would be any difference in that behavior).

What kind of responses do you give alexa to be the passing along the response as a string?

**Comment by dreimer1986** [comment]
I don't trigger anything. I get some voice from Alexa and reply to it. This then comes up in my HA Telegram bot.
Screenshot_20230627-190015

**Comment by DEADSEC-SECURITY** [comment]
> Thanks for the repy @dreimer1986 !
> 
> how do you trigger Alexa to use the "StringIntent" when responding to Alexa? Maybe I have a conceptual problem in understanding - but I thought the lambda-function/Alexa-service decides itself on the "intent to use". So a human response with "yes" or "yeah, I would like that ..." would result in a YesIntent.
> 
> What would trigger this "custom StringIntent"? I was trying to respond with my hair-color, questions, short sentences and all kinds of other things (in german, but also tried in english, despite using the German interaction model).
> 
> I can trigger the "DataTimeIntent" why saying eg. "tomrrow at one pm". I have problems triggering the "Number"-Intent as well (attempting to trigger the NumericIntentHandler) - it does not work with only responding with a number, eg. "42" (which worked in 0.8.2 for some reason - but I do not see why there would be any difference in that behavior).
> 
> What kind of responses do you give alexa to be the passing along the response as a string? Or how do you define in the initial API request to the Alexa Skill , that the human response is expected to be of type "StringIntent" (if that is even possible)?

Well in theory if you say anything that Alexa doesnt consider to be other type of intent then it will StringIntent.

**Comment by rwunsch** [comment]
Hello @dreimer1986, Hello @DEADSEC-SECURITY ,

thanks for your responses. 
I started changing things in the lambda function - and suddenly it started working as stated by @dreimer1986 .
I guess I either did not deploy the change/update to 0.9.1, or i was not patient enough after deploying - getting wrong results.
Apologies for the extra effort in answering for you guys here, and thanks for this wonderful change!

I can now combine this Skill with the [chatGPT HA-Integration](https://github.com/jjbankert/hass_chatgpt), and have discuss and conversations  with chatGPT :) 
That is sooo cool!

My "Statement" given through Alexa (alexa-actions):
image

And the chatGPT response (German, as I configured it that way - but chatGPT understood the english question :) ):
image

Thanks a TON for your help , and for this fabulous integration and Alexa Skill!
I see sooooo many benefits in this!

**Comment by DEADSEC-SECURITY** [comment]
> Hello @dreimer1986, Hello @DEADSEC-SECURITY ,
> 
> thanks for your responses. 
> I started changing things in the lambda function - and suddenly it started working as stated by @dreimer1986 .
> I guess I either did not deploy the change/update to 0.9.1, or i was not patient enough after deploying - getting wrong results.
> Apologies for the extra effort in answering for you guys here, and thanks for this wonderful change!
> 
> I can now combine this Skill with the [chatGPT HA-Integration](https://github.com/jjbankert/hass_chatgpt), and have discuss and conversations  with chatGPT :) 
> That is sooo cool!
> 
> My "Statement" given through Alexa (alexa-actions):
> image
> 
> And the chatGPT response (German, as I configured it that way - but chatGPT understood the english question :) ):
> image
> 
> Thanks a TON for your help , and for this fabulous integration and Alexa Skill!
> I see sooooo many benefits in this!

Thats awesome dude. If you ever want to open a discussion in this project and share that project of yours feel free to do so. Seems very interesting.

**Comment by dreimer1986** [comment]
Yes, I forgot to tell you that StringIntentHandler is sort of a fallback that sends back whatever does not fit elsewhere. If I reply with "Ja" to something I pass to Alexa you of course get YesIntent as answer as it succeeds way before StringIntentHandler even is asked for help.

Tbh I thought about some kind of tinkering that way, too. Nice that you already made it already and tested in german, too ^^

Currently I just use it to "scare" my parents when they show up in my flat when I am gone for work. ^^ Combined with cameras and motion detection it's a powerful tool indeed ...

Looking forward to see more of this! All those skills that utilize ChatGPT on Alexa are kinda strange to use, so this sounds like a perfect solution.

**Comment by rwunsch** [comment]
Hi @dreimer1986 ,

here is the current version of my "alexa-actions && chatGPT" NodeRed "flow":
`[{"id":"2c541c1f78680e7e","type":"tab","label":"alexa-actions-chatGPT","disabled":false,"info":"","env":[]},{"id":"ed1a06828bcb07b3","type":"inject","z":"2c541c1f78680e7e","name":"","repeat":"","crontab":"42 19 * * *","once":false,"onceDelay":0.1,"topic":"","payload":"true","payloadType":"bool","x":110,"y":80,"wires":[["4a5fce630b9929a8"]]},{"id":"4a5fce630b9929a8","type":"api-call-service","z":"2c541c1f78680e7e","name":"Alexa-actions - Which question would you like to ask?","server":"5a0908cf.d3bd78","version":5,"debugenabled":false,"domain":"script","service":"activate_alexa_actionable_notification","areaId":[],"deviceId":[],"entityId":[],"data":"{\"text\":\"Welche Frage soll ich dir beantworten?\",\"event_id\":\"actionable_notification_chatgpt\",\"alexa_device\":\"media_player.alexa_buro\",\"suppress_confirmation\":\"true\"}","dataType":"json","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","output_location":"","output_location_type":"none","x":400,"y":80,"wires":[[]]},{"id":"b65ba4f9705b78a8","type":"comment","z":"2c541c1f78680e7e","name":"Initiate alexa-actions with initial question","info":"","x":170,"y":40,"wires":[]},{"id":"1b6a9ecefa9c080d","type":"server-events","z":"2c541c1f78680e7e","name":"All chatgpt_tts events","server":"5a0908cf.d3bd78","version":2,"eventType":"chatgpt_tts","exposeToHomeAssistant":false,"eventData":"","haConfig":[{"property":"name","value":""},{"property":"icon","value":""}],"waitForRunning":true,"outputProperties":[{"property":"payload","propertyType":"msg","value":"","valueType":"eventData"},{"property":"topic","propertyType":"msg","value":"$outputData(\"eventData\").event_type","valueType":"jsonata"}],"event_type":"","x":120,"y":380,"wires":[["d8ab4ce8a2414904","4e1a2a1d85d6144c"]]},{"id":"d8ab4ce8a2414904","type":"function","z":"2c541c1f78680e7e","name":"Covert return \"content\" from ChatGPT","func":"var myContent = JSON.stringify(msg.payload.event.content);\nmsg.payload.event.content = myContent.replace(/\\\\n/g, \"\\\\n\")\n    .replace(/\\\\'/g, \"\\\\'\")\n    .replace(/\"/g, \"'\")\n    .replace(/\\\\&/g, \"\\\\&\")\n    .replace(/\\\\r/g, \"\\\\r\")\n    .replace(/\\\\t/g, \"\\\\t\")\n    .replace(/\\\\b/g, \"\\\\b\")\n    .replace(/\\\\f/g, \"\\\\f\");\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":450,"y":360,"wires":[["9547428731a97a5d"]]},{"id":"c3f9491866db978e","type":"switch","z":"2c541c1f78680e7e","name":"Notification Check - is for ChatGPT?","property":"payload.event.event_id","propertyType":"msg","rules":[{"t":"eq","v":"actionable_notification_chatgpt","vt":"str"},{"t":"eq","v":"actionable.skill.test","vt":"str"}],"checkall":"true","repair":false,"outputs":2,"x":510,"y":160,"wires":[["3248ccb29ecf867b"],["3248ccb29ecf867b"]]},{"id":"a3f3dd146d4b958a","type":"server-events","z":"2c541c1f78680e7e","name":"Alexa-actions - User Response","server":"5a0908cf.d3bd78","version":2,"eventType":"alexa_actionable_notification","exposeToHomeAssistant":false,"eventData":"","haConfig":[{"property":"name","value":""},{"property":"icon","value":""}],"waitForRunning":true,"outputProperties":[{"property":"payload","propertyType":"msg","value":"","valueType":"eventData"},{"property":"topic","propertyType":"msg","value":"$outputData(\"eventData\").event_type","valueType":"jsonata"},{"property":"event_type","propertyType":"msg","value":"$outputData(\"eventData\").event_type","valueType":"jsonata"}],"x":130,"y":160,"wires":[["c3f9491866db978e","4f572ac4c9445f9f"]]},{"id":"3248ccb29ecf867b","type":"switch","z":"2c541c1f78680e7e","name":"Notification Response","property":"payload.event.event_response_type","propertyType":"msg","rules":[{"t":"eq","v":"ResponseYes","vt":"str"},{"t":"eq","v":"ResponseNo","vt":"str"},{"t":"eq","v":"ResponseString","vt":"str"},{"t":"eq","v":"ResponseNumeric","vt":"str"},{"t":"eq","v":"ResponseNone","vt":"str"},{"t":"empty"}],"checkall":"true","repair":false,"outputs":6,"x":580,"y":240,"wires":[["c7cb14cb53d9c6e6"],["c7cb14cb53d9c6e6"],["c7cb14cb53d9c6e6"],["c7cb14cb53d9c6e6"],[],[]]},{"id":"6f15330b63399af1","type":"api-call-service","z":"2c541c1f78680e7e","name":"ChatGPT-Frage","server":"5a0908cf.d3bd78","version":5,"debugenabled":true,"domain":"chatgpt","service":"chat","areaId":[],"deviceId":[],"entityId":[],"data":"{\"messages\":[{\"role\":\"system\",\"content\":\"Sprich mich immer mit DU an, und nicht mit SIE. Benutze nicht mehr als 30 Worte. Beende deine Antwort mit einer offenen Frage, wie zB. ob ich über den Inhalt der ürsprünglichen Frage noch mehr wissen möchte.  {{payload.custom_data.additional_information}}\"},{\"role\":\"user\",\"content\":\"{{payload.event.event_response}}\"}],\"callback_event\":\"chatgpt_tts\"}","dataType":"json","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":1080,"y":260,"wires":[["606f601dc6ebd98d"]]},{"id":"9547428731a97a5d","type":"api-call-service","z":"2c541c1f78680e7e","name":"Alexa-actions - answer and next question","server":"5a0908cf.d3bd78","version":5,"debugenabled":true,"domain":"script","service":"activate_alexa_actionable_notification","areaId":[],"deviceId":[],"entityId":[],"data":"{\"text\":\"{{payload.event.content}}\",\"event_id\":\"actionable_notification_chatgpt\",\"alexa_device\":\"media_player.alexa_buro\",\"suppress_confirmation\":\"true\"}","dataType":"json","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","output_location":"","output_location_type":"none","x":800,"y":380,"wires":[["9e992e58d6ad674c"]]},{"id":"4f572ac4c9445f9f","type":"debug","z":"2c541c1f78680e7e","name":"DEBUG - Alexa-actions-user-response","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":250,"y":240,"wires":[]},{"id":"606f601dc6ebd98d","type":"debug","z":"2c541c1f78680e7e","name":"DEBUG - ChatGPT question","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":1220,"y":320,"wires":[]},{"id":"4e1a2a1d85d6144c","type":"debug","z":"2c541c1f78680e7e","name":"DEBUG - ChatGPT Event","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":390,"y":420,"wires":[]},{"id":"9e992e58d6ad674c","type":"debug","z":"2c541c1f78680e7e","name":"DEBUG - actions-answer-and-question","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":850,"y":440,"wires":[]},{"id":"c7cb14cb53d9c6e6","type":"function","z":"2c541c1f78680e7e","name":"add additional information.","func":"/*\nArray.prototype.findReg = function (match) {\n    var reg = new RegExp(match);\n\n    return this.filter(function (item) {\n        return typeof item == 'string' && item.match(reg);\n    });\n}\n\n//var y = x.findReg(\"temperature\");\n*/\n\nvar additional_information;\nvar date;\nvar temperature_outside;\nmsg.payload.custom_data = {};\n\n\ndate = new Date().toLocaleString(\"de-DE\", { dateStyle: \"long\", timeStyle: \"long\", timeZone: \"Europe/Berlin\" });\nadditional_information = \"Mein Datum und Uhrzeit: \" + date + \" .\";\n\nadditional_information += \"Mein Standort: ACME-Street 1A, 12345 AMCE-CITY, Deutschland. \";\n\ntemperature_outside = global.get(\"homeassistant\").homeAssistant.states['sensor.pool_temperature_sensor_outside_temperatur'].state;\nadditional_information += \"Meine Aussentemperatur: \" + temperature_outside + \"°.\";\n\nmsg.payload.custom_data.additional_information = additional_information;\n\nmsg.payload.custom_data.all_information = global.get(\"homeassistant\").homeAssistant.states;\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":840,"y":260,"wires":[["4bc96a3e9326501d","6f15330b63399af1"]]},{"id":"4bc96a3e9326501d","type":"debug","z":"2c541c1f78680e7e","name":"DEBUG - additional information","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":910,"y":320,"wires":[]},{"id":"5a0908cf.d3bd78","type":"server","name":"Home Assistant","version":5,"addon":true,"rejectUnauthorizedCerts":true,"ha_boolean":"y|yes|true|on|home|open","connectionDelay":true,"cacheJson":true,"heartbeat":false,"heartbeatInterval":"30","areaSelector":"friendlyName","deviceSelector":"friendlyName","entitySelector":"friendlyName","statusSeparator":"at: ","statusYear":"hidden","statusMonth":"short","statusDay":"numeric","statusHourCycle":"h23","statusTimeFormat":"h:m","enableGlobalContextStore":true}]`

This is STILL WORK IN PROGRESS! 
(and a lot in german)

I am using this ChatGPT integration, which does not do much more than accessing the openAI-API, authenticates, and receives the response synchronously, and triggers an event: https://github.com/jjbankert/hass_chatgpt

I still want/need to:

1. save previous questions and responses and send them through with the next request to chatGPT.
2. add some more "addintional information", and find out how to query specific entities on a request for HA-information, such as " what is the Outside temperature", or "is the light in the kitchen on" ... or "turn off all lights"....
(I CAN do some of that directly with the NabuCasa-alexa-intengration - but I would like to have a more "natural" conversation with Alexa, using ChatGPT as the actual AI).
3. make this Skill (alexa-actions) to always use the intital question from HA, even when calling it with "Alexa, open <initiation name of this skill, for me "home-assistant", default: alexa-actions or proactive notifications>". 
4. Find out which ALEXA device is initiating the skill, for this one to be the target of subsequent "alexa-media-player/alexa-actions"-responses.

Hope this give an idea how I use it :) - and maybe you find some better ways than what I do currently.

**Pull request merged**